source/uwp/Renderer/lib/AdaptiveActionsConfig.h

```diff
@@ -36,7 +36,13 @@ namespace AdaptiveCards { namespace Uwp
         IFACEMETHODIMP put_Spacing(ABI::AdaptiveCards::Uwp::Spacing value);
 
     private:
-        AdaptiveCards::ActionsConfig m_sharedActionsConfig;
+        ABI::AdaptiveCards::Uwp::ActionAlignment m_actionAlignment;
+        ABI::AdaptiveCards::Uwp::ActionsOrientation m_actionsOrientation;
+        UINT m_buttonSpacing;
+        UINT m_maxActions;
```

Comment:
>UINT m_maxActions; [](start = 8, length = 18)

MaxActions and all these other properties aren't going to have the correct default values when you initialize these classes using the parameterless constructor, right?

That would mean hosts using the programmatic way of constructing HostConfig MUST specify every single property? Otherwise they're going to end up with values like a max of 0 actions?

I think the correct default values should be assigned when you new up each config object. So maybe that'd involve newing up a shared config object in the parameterless constructor and then calling the other constructor with that new shared object?

Reply:
Good point. I'll fix this.

---
In reply to: [145463181](https://github.com/Microsoft/AdaptiveCards/pull/778#discussion_r145463181) [](ancestors = 145463181)